handler/src/main/java/io/netty/handler/traffic/TrafficCounter.java

```diff
@@ -263,6 +237,26 @@ synchronized void resetAccounting(long newLastTime) {
      * Constructor with the {@link AbstractTrafficShapingHandler} that hosts it, the Timer to use, its
      * name, the checkInterval between two computations in millisecond.
      *
+     * @param executor
+     *            the underlying executor service for scheduling checks, might be null when used
+     * from {@link GlobalChannelTrafficCounter}.
+     * @param name
+     *            the name given to this monitor.
+     * @param checkInterval
+     *            the checkInterval in millisecond between two computations.
+     */
+    public TrafficCounter(ScheduledExecutorService executor, String name, long checkInterval) {
+        trafficShapingHandler = null;
+        this.executor = executor;
+        this.name = name;
+        init(checkInterval);
+
+    }
+
+    /**
+     * Constructor with the {@link AbstractTrafficShapingHandler} that hosts it, the Timer to use, its
```

Comment:
I think `the Timer` is not correct as we not use a Timer anymore but a ScheduledExecutorService